src/analyze/expr.rs

```diff
@@ -1270,10 +1315,18 @@ impl Expr {
                     return expr;
                 }
             }
-            error_handler.error(
-                SemanticError::InvalidCast(expr.ctype.clone(), ctype.clone()),
-                expr.location,
-            );
+            // There is probably a better way to do this
+            // check not type error and that if rval is a pointer, it is not a type error
+            if *ctype != Type::Error {
+                if let Type::Pointer(a, _) = expr.ctype.clone() {
+                    if *a != Type::Error {
+                        error_handler.error(
+                            SemanticError::InvalidCast(expr.ctype.clone(), ctype.clone()),
+                            expr.location,
+                        );
+                    }
+                }
```

Comment:
This looks like the incorrect logic, if it's not a pointer you should still report the error. Is there a reason you're special casing pointers here?

Reply:
This does seem incorrect, I don't remember exactly why I chose to do this. I think it got rid of a cascading error in one case, but the nested if hides an invalid cast for nonpointers like you say. For now I can just remove the pointer part